File: backend-java/src/main/java/com/carrental/model/Booking.java
package com.carrental.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;

@Data
@NoArgsConstructor
@Document(collection = "bookings")
public class Booking {

    @Id
    private String id;
    
    private String customerId; // User ObjectId
    
    private String carId; // Car ObjectId
    
    private String ownerId; // User ObjectId

    private String customerName;
    private String customerPhone;
    private String customerEmail;
    
    private String carName;
    private String carNumber;
    
    private String ownerName;
    private String ownerEmail;

    private Date startDate;
    
    private Date endDate;
    
    private Double totalPrice;
    
    private String paymentMethod; // Card, UPI, Cash, Razorpay
    
    private String paymentStatus = "Pending"; // Pending, Completed, Failed
    
    private String status = "Pending"; // Pending, Confirmed, Completed, Cancelled, Rejected
    
    private String notes;
    
    private Date createdAt = new Date();
}

File: backend/models/Booking.js
const mongoose = require('mongoose');

const bookingSchema = new mongoose.Schema({
  customer: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  car: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Car',
    required: true
  },
  owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  startDate: {
    type: Date,
    required: [true, 'Please provide start date']
  },
  endDate: {
    type: Date,
    required: [true, 'Please provide end date']
  },
  totalPrice: {
    type: Number,
    required: true,
    min: 0
  },
  paymentMethod: {
    type: String,
    required: true,
    enum: ['Card', 'UPI', 'Cash', 'Razorpay']
  },
  paymentStatus: {
    type: String,
    enum: ['Pending', 'Completed', 'Failed'],
    default: 'Pending'
  },
  status: {
    type: String,
    enum: ['Pending', 'Confirmed', 'Completed', 'Cancelled', 'Rejected'],
    default: 'Pending'
  },
  notes: {
    type: String
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Booking', bookingSchema);
